telosnetwork/telos-wallet
# auto-close dialog when no access to the camera

# Fixes #29 

## Description
The dialog did not react to the denial of access to the camera remaining open and useless. The user had to refresh to come back to the balances page. This PR fixes that idle behavior by closing the dialog when non-access to the camera is detected, showing to the user an appropriate notification message.

## Test notes
To test this new behavior you need to follow these steps:
- clean any pre-set configuration to access the camera (forget your previous answer)
- Click on the QR button (center between **Send** and **Receive** buttons)
- A dialog opens with a loading spinner (this is new)
- A browser dialog prompts if you allow the site to access the camera.
- Click on Deny.
- The dialog should close automatically and an error message should be shown asking for approval.


##screenshots

![image](https://user-images.githubusercontent.com/4420760/210201390-05524c9b-efa4-4d32-99ab-597b126a13dd.png)

![image](https://user-images.githubusercontent.com/4420760/210201096-093cd246-a414-4462-8d7d-eeef35a753f4.png)


## Checklist:
